src/application.ts

```diff
@@ -0,0 +1,187 @@
+import { Value } from "./parts/value.class";
+import { DataBind } from "./data-bind.class";
+import { ActionButton, NumberButton } from "./parts/button.class";
+import { CustomCollection } from "./parts/custom-collection";
+import {
+    OperationHistory,
+    OperationHistoryType,
+} from "./parts/operation-history.interface";
+import { ActionType, Action } from "./parts/action.class";
+
+export class Application {
+    private binding?: DataBind;
+    private stack: CustomCollection<OperationHistory> = new CustomCollection();
+
+    private _onNumberButtonClick: (value: Value) => void = (value) =>
+        this.onNumberButtonClick(value);
+
+    private _onActionButtonClick: (action: ActionType) => void = (action) =>
+        this.onActionButtonClick(action);
+
+    bind() {
+        if (this.binding) return;
+
+        this.binding = new DataBind();
+
+        // 各ボタンと処理するロジックを関連付ける
+        this.binding.buttons.forEach((_, element) => {
+            if (!element) {
+                return;
+            }
+            switch (element.constructor) {
+                case NumberButton:
+                    (element as NumberButton).setOnClick(
+                        this._onNumberButtonClick
+                    );
+                    break;
+                case ActionButton:
+                    (element as ActionButton).setOnClick(
+                        this._onActionButtonClick
+                    );
+                    break;
+            }
+        });
+    }
+
+    unbind() {
+        if (!this.binding) return;
+        this.binding.unbindAll();
+        this.binding = undefined;
+    }
+
+    reset(initialValue: Value = Value.ZERO) {
+        this.stack.clear();
+        this.stack.push(initialValue);
+
+        this.updateOutput(this.calculateTotal());
+    }
+
+    private updateOutput(value: Value) {
+        this.binding?.display?.setValue(value);
+    }
+
+    private calculate(valueA: Value, action: Action, valueB: Value): Value {
+        switch (action.value) {
+            case ActionType.ADD:
+                return valueA.add(valueB);
+
+            case ActionType.SUBTRACT:
+                return valueA.subtract(valueB);
+
+            case ActionType.MULTIPLE:
+                return valueA.multiple(valueB);
+
+            case ActionType.DIVIDE:
+                if (valueB.isZero()) {
+                    return new Value(0);
+                } else {
+                    return valueA.divide(valueB);
```

Comment:
May need a comment not to catch the exception,
// Not catching the exception since we are checking 0 division manually

OR remove zero check and catch the exception?